### PR TITLE
GitHub actions

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
This adds a github action to

- build
- test

Mosh on both macos and ubuntu

Currently, the macos build seems to be failing, ~~maybe due to 6fe7cde94db404e0a01a0af39cdd27c4646d067c, though this still requires further investigation~~ actually, I think the failure is due to [this protobuf commit](https://github.com/protocolbuffers/protobuf/commit/624d29d83306f3ce2c7e092dd44a891e04215e67) as discussed [here]( https://github.com/protocolbuffers/protobuf/issues/9181)

### CODE REVIEW

**Comment by ezzieyguywuf** [comment]
@achernya can you please review this PR?

File: .github/workflows/ci.yml
Comment (achernya) [inline .github/workflows/ci.yml:11]: This comment is probably unnecessary. I'd suggest omitting it.

File: .github/workflows/ci.yml
Comment (achernya) [inline .github/workflows/ci.yml:7]: Let's run on both Ubuntu 18.04 and 20.04 explicitly.

File: .github/workflows/ci.yml
Comment (achernya) [inline .github/workflows/ci.yml:20]: Is it really the case you don't need autotools here? https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-Readme.md doesn't seem to list it. I would expect you to need to explicitly install autoconf and automake.

File: .github/workflows/ci.yml
Comment (achernya) [inline .github/workflows/ci.yml:32]: For both of these, we'll want the make to run in verbose mode so we can see the full invocations. I believe you can do this by running `make V=1` in mosh's build.

File: README.md
Comment (achernya) [inline README.md:1]: Can you update the change description to someting like "Replace Travis CI with Github Actions"?

File: configure.ac
Comment (achernya) [inline configure.ac:45]: Please update the link in the change description to "https://github.com/protocolbuffers/protobuf/issues/9181" directly, Also please leave a comment here with a TODO to remove this suppression once the issue is resolved.

**Review by achernya — CHANGES_REQUESTED** [review]

File: configure.ac
Comment (ezzieyguywuf) [reply configure.ac:45]: The only reason I didn't link the protobuf issue directly is due to the spam it creates in the protobuf issue each time I force push updates.

Do you still want it linked directly in the commit message?

Or perhaps I can add the link to this PR (ie the description above) to have a record of it after this is merged?

Comment (achernya) [reply configure.ac:45]: I would prefer that there be a comment mentioning the issue/PR directly above the warning. I personally would not worry about spamming the issue -- hyperlinks and cross references exist for a reason.

File: .github/workflows/ci.yml
Comment (ezzieyguywuf) [reply .github/workflows/ci.yml]: Done.

File: .github/workflows/ci.yml
Comment (ezzieyguywuf) [reply .github/workflows/ci.yml]: Done

File: .github/workflows/ci.yml
Comment (ezzieyguywuf) [reply .github/workflows/ci.yml]: I'm not sure why the actions [ran fine][1] without these, but I went ahead and added them just to be safe - similarly to you, I had some reservations about not explicitly installing them. I was a bit confused after reading [this][2], since I thought that meant that autotools stuff was installed by default.

[1]: https://github.com/ezzieyguywuf/mosh/actions/runs/1767435406
[2]: https://github.com/actions/starter-workflows/issues/53

File: .github/workflows/ci.yml
Comment (ezzieyguywuf) [reply .github/workflows/ci.yml:30]: Done.

File: README.md
Comment (ezzieyguywuf) [reply README.md:1]: Done.

File: configure.ac
Comment (ezzieyguywuf) [reply configure.ac:47]: No problem - I added it to the commit message as well as to the comment in the file itself

**Comment by ezzieyguywuf** [comment]
@achernya I believe I've addressed all your original comments. Can you let me know if there's anything else blocking this from moving forward?

**Review by achernya — APPROVED** [review]

**Pull request merged**
